server/src/main/java/org/elasticsearch/action/admin/indices/alias/TransportIndicesAliasesAction.java

```diff
@@ -106,15 +107,19 @@ protected void masterOperation(
         Task task,
         final IndicesAliasesRequest request,
         final ClusterState state,
-        final ActionListener<AcknowledgedResponse> listener
+        final ActionListener<IndicesAliasesResponse> listener
     ) {
 
         // Expand the indices names
         List<AliasActions> actions = request.aliasActions();
         List<AliasAction> finalActions = new ArrayList<>();
+        List<AliasActionResult> actionResults = new ArrayList<>();
```

Comment:
I am a little uncomfortable with the change made in this file. Specifically, the use of the actionResult variable which is modified throughout the main loop over actions. Also, the fact the the action result is added to the final actionResults list at different places in the code. This is done because of the use of `continue`  in a few places. I think a better design would require a refactor of this loop to remove the use of `continue` statements. I was trying to avoid changing the code too much ... but kinda seems like I already passed that point.

Reply:
:+1:  #107017